src/Sylius/Component/Order/Model/Order.php

```diff
@@ -458,8 +428,6 @@ public function removeComment(CommentInterface $comment)
             $comment->setOrder(null);
```

Comment:
This should be removed as item __must__ be assigned to some order, otherwise it's broken.